lib/client.js

```diff
@@ -78,6 +81,11 @@ module.exports = function (dependencies) {
               this.config.token.regenerate(tokenGeneration);
               resolve(this.write(notification, device, retryCount + 1));
               return;
+            } else if (status === "500" && response.reason === "InternalServerError") {
+              stream.connection.close();
+              let error = new VError("Error 500, stream ended unexpectedly");
```

Comment:
Sounds great :). 

But I guess `status` would be `null` in [lib/client.js:109](https://github.com/cmdkoh/node-apn/blob/710f68c9fb8ddd9cd71b089221ff61f7cb2a8fd0/lib/client.js#L109) and [lib/client.js:115](https://github.com/cmdkoh/node-apn/blob/710f68c9fb8ddd9cd71b089221ff61f7cb2a8fd0/lib/client.js#L115), because there would no response from server on stream error event. Am I right?

BTW, I'm not a maintainer, so you guys (@florianreinhart and @jpwilliams ) have to decide how to do. :)